File: src/main/java/com/dude/dms/startup/StartUpRunner.java
package com.dude.dms.startup;

import com.dude.dms.backend.brain.polling.DocPollingService;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.boot.CommandLineRunner;
import org.springframework.context.i18n.LocaleContextHolder;
import org.springframework.stereotype.Component;

import java.io.File;
import java.io.IOException;
import java.util.Locale;

import static com.dude.dms.backend.brain.OptionKey.LOCALE;

@Component
public class StartUpRunner implements CommandLineRunner {

    private static final Logger LOGGER = LoggerFactory.getLogger(StartUpRunner.class);

    @Autowired
    private DocPollingService docPollingService;

    @Autowired
    private UpdateChecker updateChecker;

    @Autowired
    private DirectoryManager directoryManager;

    @Autowired
    private DemoDataManager demoDataManager;

    @Override
    public void run(String... args) throws IOException {
        updateChecker.check();
        createOptionsFile();
        directoryManager.checkDirectories();
        demoDataManager.createDemoData();
        docPollingService.manualPoll();
        LocaleContextHolder.setLocale(Locale.forLanguageTag(LOCALE.getString()));
    }

    private static void createOptionsFile() throws IOException {
        File prop = new File("options.properties");
        if (!prop.exists()) {
            LOGGER.info("Creating user properties...");
            prop.createNewFile();
        }
    }
}

File: src/main/java/com/dude/dms/startup/StartUpRunner.kt
package com.dude.dms.startup

import com.dude.dms.backend.service.DocService
import com.dude.dms.brain.DmsLogger
import com.dude.dms.brain.FileManager
import com.dude.dms.brain.options.Options
import com.dude.dms.brain.polling.DocImportService
import com.dude.dms.brain.t
import com.dude.dms.updater.UpdateChecker
import org.springframework.boot.CommandLineRunner
import org.springframework.context.i18n.LocaleContextHolder
import org.springframework.stereotype.Component
import java.util.*

@Component
class StartUpRunner(
        private val docService: DocService,
        private val fileManager: FileManager,
        private val docImportService: DocImportService,
        private val updateChecker: UpdateChecker,
        private val directoryChecker: DirectoryChecker,
        private val optionsChecker: OptionsChecker
) : CommandLineRunner {

    override fun run(vararg args: String) {
        updateChecker.check(true)
        optionsChecker.checkOptions()
        directoryChecker.checkDirectories()
        LOGGER.info("Cleaning up files...")
        fileManager.getAllPdfs().forEach { file -> if (docService.findByGuid(file.name.takeWhile { it != '.' }) == null) file.delete() }
        fileManager.getAllImages().forEach { file -> if (docService.findByGuid(file.name.takeWhile { it != '_' }) == null) file.delete() }

        // Migration 0.0.2 -> 0.0.3
        docService.findIncomplete().forEach {
            if (it.deleted == null) {
                it.deleted = false
                docService.save(it)
            }
        }

        Thread { docImportService.import() }.start()
        LocaleContextHolder.setLocale(Locale.forLanguageTag(Options.get().view.locale))
        LOGGER.info(t("startup.complete"))
    }

    companion object {
        private val LOGGER = DmsLogger.getLogger(StartUpRunner::class.java)
    }
}